modules/core/migrations/v7/genesis.go

```diff
@@ -0,0 +1,40 @@
+package v7
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	genutiltypes "github.com/cosmos/cosmos-sdk/x/genutil/types"
+
+	clientv7 "github.com/cosmos/ibc-go/v6/modules/core/02-client/migrations/v7"
+	host "github.com/cosmos/ibc-go/v6/modules/core/24-host"
+	"github.com/cosmos/ibc-go/v6/modules/core/types"
+)
+
+// MigrateGenesis accepts an exported IBC client genesis file and migrates it to:
+//
+// - Update solo machine client state protobuf definition (v2 to v3)
+// - Remove all solo machine consensus states
+// - Remove any localhost clients
+func MigrateGenesis(appState genutiltypes.AppMap, cdc codec.ProtoCodecMarshaler) (genutiltypes.AppMap, error) {
+	if appState[host.ModuleName] != nil {
+		// ensure legacy solo machines are registered
+		clientv7.RegisterInterfaces(cdc.InterfaceRegistry())
+
+		// unmarshal relative source genesis application state
+		ibcGenState := &types.GenesisState{}
+		cdc.MustUnmarshalJSON(appState[host.ModuleName], ibcGenState)
```

Comment:
[q] is it safe to panic during migrations like this?

Reply:
Yes it is, this is being run while the chain is halted

Reply:
thanks for the clarification!